### DISCUSSION TITLE
API Ideas

### DISCUSSION THREAD

**Discussion by Waidhoferj** [y-crdt/ypy#48] [Ideas]
Thread to track API changes, additions and updates.

**Comment by Waidhoferj** [comment]
# `YArray` API Modifications
We could change the `YArray` API in the following ways to make it more Pythonic:
## Replace `push(txn, items: List)` with `extend(txn, items: Iterable)`
**Pros**:
- This mirrors the Python List API which makes the function more discoverable
- Works with all iterables instead of just lists

**Cons**:
- Breaking change could require refactoring dependent codebases

## Add an `append(txn, item)` method
**Pros**:
- Allows for single insertion without wrapping it in an array. If you pass an array to the current push function, expecting the array to be inserted, you'll actually end up with a flattened array. I found this a bit counterintuitive, so making a single element `append` would allow for some distinction.

**Cons**:
- Diverges from the Y.js and Ywasm APIs

**Reply by davidbrochart** [reply]
I think `extend` and `append` are better names in the Python world too. We already diverge from the Yjs API with the transaction context manager. I think it's fine to better adhere to the language, rather than sticking to a universal API.
